src/test/ui/async-await/dont-suggest-missing-await.stderr

```diff
@@ -2,7 +2,7 @@ error[E0308]: mismatched types
   --> $DIR/dont-suggest-missing-await.rs:14:18
    |
 LL | async fn make_u32() -> u32 {
-   |                        --- checked the `Output` of this `async fn`, found opaque type
+   |                        --- async functions return futures
```

Comment:
Love the new output! It is certainly better.

I wonder if we should (maybe in the future?) give output similar to "async function `make_u32` returns an `impl Future<Output = u32>`". I know I've pushed for that in the past and not everyone is happy with that, but I fear that we are not taking advantage of teaching people about what "futures" are at the type level. Having said that, we could reserve a more "accurate" phrasing for specific cases (like when we suggest appending `.await` to a function call), and maybe make it part of a `note`.

Reply:
I'd forgotten I had https://github.com/rust-lang/rust/pull/87673 open that touches the same general place, mainly moving the `span_label` to a `span_note` using a `MultiSpan` to have _more_ text. I wonder if instead of landing my PR you could incorporate some of the changes I made?

I just spent a few minutes trying to accomplish what you mentioned (use the name of the method) and this is what I landed at, which I feel looks "pretty neat", but I want us to go for "easy to understand by newcomers":

```
error[E0308]: mismatched types
  --> src/test/ui/async-await/dont-suggest-missing-await.rs:14:18
   |
14 |         take_u32(x)
   |                  ^ expected `u32`, found opaque type
   |
note: while checking the return type of the async function `make_u32`
  --> src/test/ui/async-await/dont-suggest-missing-await.rs:7:24
   |
7  | async fn make_u32() -> u32 {
   |          --------      ^^^ the desugared version of `make_u32` returns `impl Future<Output = u32>`
   |          |
   |          async functions return futures
   = note:     expected type `u32`
           found opaque type `impl Future`
help: consider `await`ing on the `Future`
   |
14 |         take_u32(x.await)
   |                   ^^^^^^
```
<img width="739" alt="Screen Shot 2021-08-02 at 7 36 59 AM" src="https://user-images.githubusercontent.com/1606434/127879005-34129970-020b-4a37-99e7-7ac8ca5af54a.png">

I'll push the code to #87673 in a minute and you can tell me if you can pick from it so you can run with it and make it your own if there are more things we want to do here :)

Reply:
Oh! While I was getting this working, the PR got approved and selected in a rollup (otherwise I'd kill it). We'll have to rebase _this_ PR after that lands, sorry about that :-/

The follow up code the above screenshot came from is at https://github.com/rust-lang/rust/commit/bfcaf8be327826cf32e861e66a90d34e98b02f3d. Would you have the time/desire to take that and make it the best output it can be?

Reply:
Sure! I likely won't have time until next weekend, but I would be happy to! I'll likely open a new PR with that change.

Reply:
And ty so much for the help!